multithreaded/Dfit_2D_main.cpp

```diff
@@ -289,73 +378,117 @@ int main(int argc, char *argv[])
                                 std::cerr<< "Error in substituting parameters param " << rep_var.second << " not found.\n";
 				return 1;
                         }
-		replaceIndexVect.push_back(std::make_pair(index_replaced, index_substitute));
+			replaceIndexVect.push_back(std::make_pair(index_replaced, index_substitute));
 		}
 
 
 
-	// Start the minimization
-	// Define a fit function for Minuit
-	auto fchi2 = wrap_chi2(D_PDFs_get, B_PDFs_get, vect_2D, MC_MD, dMC_MD, MC_MB, dMC_MB, replaceIndexVect);
-	ROOT::Math::Functor f(fchi2, (nvar_md + nvar_mb) * ncontr + ncontr - 1);
-	//ROOT::Math::Functor f(fchi2, (nvar_md + nvar_mb) * ncontr + ncontr);
-	min->SetFunction(f);
-	min->Minimize();
+		// Start the minimization
+		// Define a fit function for Minuit
+		auto fchi2 = wrap_chi2(D_PDFs, B_PDFs, vect_2D, Config::MC_MD, Config::dMC_MD, Config::MC_MB, Config::dMC_MB, replaceIndexVect, int_choose_fit, hist2D);
+		ROOT::Math::Functor f(fchi2, n_all);
+		//ROOT::Math::Functor f(fchi2, (nvar_md + nvar_mb) * ncontr + ncontr);
+		min->SetFunction(f);
+		min->Minimize();
 
-	// Print the fit results
-	std::ofstream results(TString::Format("results_%d.txt", sign));
-	results << min->Status() << "  " << min->MinValue() << std::endl;
-	for (int i = 0; i < (nvar_md + nvar_mb) * ncontr + ncontr - 1; i++)
-	//for (int i = 0; i < (nvar_md + nvar_mb) * ncontr + ncontr; i++)
-	{
-		results << min->X()[i] << "  " << min->Errors()[i] << std::endl;
-	}
-	results.close();
-	return 0;
-}
-std::function<double(const double*)> wrap_chi2(const std::vector<PDFInterface*>& D_PDFs_get, const std::vector<PDFInterface*>& B_PDFs_get, std::vector<std::pair<double, double>> vect_2D, const std::vector<std::vector<double>>& MC_MD, const std::vector<std::vector<double>>& dMC_MD, const std::vector<std::vector<double>>& MC_MB, const std::vector<std::vector<double>>& dMC_MB, const std::vector<std::pair<int, int>>& replaceIndexVect){
+		// Print the fit results
+		TString path, path1;
+	        if (Config::binned){
+			path1 = "results_binned100kMU_2";
+			path = TString::Format("results_binned100kMU_2/fit2D_%d", itry);
+		}
+		else{ 
+			path1 = "results_unbinned100kMU_2";
+			path = TString::Format("results_unbinned100kMU_2/fit2D_%d", itry);
+		}			
+		gSystem->Exec(TString::Format("mkdir %s", path1.Data()).Data());
+		gSystem->Exec(TString::Format("mkdir %s", path.Data()).Data());
+		std::ofstream results(TString::Format("%s/results_%d_%d.txt", path.Data(), Config::sign, int_choose_fit));
+		std::cout<<std::setprecision(25);
+		results << min->Status() << "  " << min->MinValue() << std::endl;
+		std::cout<<std::setprecision(10);
+		if (min->Status()!=0){// || min->MinValue()>24.8e6){
+			std::cout << "Bad status of fit "<< min->Status() << " chi2 is " << min->MinValue() << std::endl;
+			previous_fit = false; 
```

Comment:
Wydaje mi się że powinieneś ustawić `goodfit = false;` inaczej jeśli wcześniejszy fit w łańcuszku się skończy poprawnie to cały jest oznaczony jako OK.